app/models/spree/order/checkout.rb

```diff
@@ -67,12 +67,6 @@ def self.define_state_machine!
                 transition to: :cart, unless: :completed?
               end
 
-              if states[:payment]
-                before_transition to: :complete do |order|
-                  order.process_payments! if order.payment_required?
```

Comment:
I'm not sure we can remove this..?

Reply:
Yeah, it was definitely a "let's take this out and see what breaks" commit. I'm going to look through today and see where we'd need to change things and if it's manageable. It would be really nice to separate the two things and remove a callback.

Reply:
Decoupling payment processing from order completion might end up being a pretty big refactor. Maybe it's worth doing it in a separate PR?